### ISSUE TITLE
Enhance a11y for `Checkbox`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#5397]
name: Checkbox

### Description
1 accessibility violation was found in the `Checkbox`.

#### [Components / Checkbox / Indeterminate](https://yamada-ui.github.io/yamada-ui/?path=/story/components-checkbox--indeterminate)

Received:
ARIA attributes must be used as specified for the element's role (aria-conditional-attr)

Fix all of the following:
  Remove aria-checked, or set it to "false" to match the real checkbox state

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/aria-conditional-attr?application=axeAPI

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.
